[Bug]: AMMs hit `panic: missing min price in levels`

### Problem encountered

Running market-sim AMM scenarios against [b1cd53a](https://github.com/vegaprotocol/vega/commit/b1cd53abe0fe4403fbbf0b791d4cb315a732cd1f) we hit panic.

Market-sim dump including logs and replay file:
[vega-sim-sri7j_u7 2.zip](https://github.com/user-attachments/files/15550980/vega-sim-sri7j_u7.2.zip)

```
panic: missing min price in levels

goroutine 52521 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x40?, 0x1?, {0x1?, 0x0?, 0x0?})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:196 +0x78
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0x14004a992b0, {0x14009841840, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:262 +0x2c0
go.uber.org/zap.(*Logger).Panic(0x62c45dad955c4ec?, {0x107e3ea1f?, 0x0?}, {0x14009841840, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.25.0/logger.go:258 +0x54
code.vegaprotocol.io/vega/core/matching.(*IndicativePriceAndVolume).getLevelsWithinRange(0x14004ad2120, 0x14006778840, 0x14006778860)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/indicative_price_and_volume.go:318 +0x3a8
code.vegaprotocol.io/vega/core/matching.(*IndicativePriceAndVolume).GetCumulativePriceLevels(0x14004ad2120, 0x104dee014?, 0x140067784e0)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/indicative_price_and_volume.go:346 +0x194
code.vegaprotocol.io/vega/core/matching.(*OrderBook).buildCumulativePriceLevels(0x14002af0e60)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/orderbook.go:536 +0x140
code.vegaprotocol.io/vega/core/matching.(*OrderBook).GetIndicativePrice(0x14002af0e60)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/orderbook.go:448 +0x28
code.vegaprotocol.io/vega/core/matching.(*CachedOrderBook).GetIndicativePrice(0x14002f0eea0)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/cached_orderbook.go:226 +0x64
code.vegaprotocol.io/vega/core/execution/future.(*Market).getAuctionPrice(0x14002af9c00)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/isolated_margin.go:79 +0x94
code.vegaprotocol.io/vega/core/execution/future.(*Market).marginsAuction(0x14002af9c00, {0x1091fc4d8, 0x140023dc990}, 0x14004a99520)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/margin.go:67 +0x168
code.vegaprotocol.io/vega/core/execution/future.(*Market).calcMargins(0x14002af9c00, {0x1091fc4d8, 0x140023dc990}, 0x1400526fb00?, 0x40?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/margin.go:30 +0x58
code.vegaprotocol.io/vega/core/execution/future.(*Market).checkMarginForOrder(0xfffffffffffff243?, {0x1091fc4d8, 0x140023dc990}, 0x1400526fb00?, 0x40?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:3261 +0x28
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitValidatedOrder(0x14002af9c00, {0x1091fc4d8?, 0x140023dc990}, 0x14004a99520)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:2453 +0x468
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitOrder(0x14002af9c00, {0x1091fc4d8, 0x140023dc990}, 0x14004a99520)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:2387 +0x194
code.vegaprotocol.io/vega/core/execution/future.(*Market).SubmitOrderWithIDGeneratorAndOrderID(0x14002af9c00, {0x1091fc4d8?, 0x140023dc990}, 0x40?, {0x1400bee2840, 0x40}, {0x1091cb8c0?, 0x14003ff6090?}, {0x14007341b40, 0x40}, ...)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:2346 +0x50c
code.vegaprotocol.io/vega/core/execution.(*Engine).SubmitOrder(0x14000830c00, {0x1091fc4d8, 0x140023dc990}, 0x1400bca76e0, {0x1400bee2840, 0x40}, {0x1091cb8c0, 0x14003ff6090}, {0x14007341b40, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/engine.go:947 +0x380
code.vegaprotocol.io/vega/core/processor.(*BMIProcessor).ProcessBatch(0x14006761940, {0x1091fc4d8, 0x140023dc990}, 0x14004ac23c0, {0x1400bee2840, 0x40}, {0x14007341980, 0x40}, {0x1092522a8, 0x1400067f740})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/batch_market_instructions_processor.go:237 +0x1040
code.vegaprotocol.io/vega/core/processor.(*App).DeliverBatchMarketInstructions(0x14000a8f500, {0x1091fc4d8, 0x140023dc990}, {0x109228fc0, 0x14009840500}, {0x14007341980, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:1670 +0xfc
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func50({0x1091fc4d8, 0x140023dc990}, {0x109228fc0, 0x14009840500})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:571 +0x64
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).CheckBatchMarketInstructionsW.func51({0x1091fc4d8, 0x140023dc990}, {0x109228fc0, 0x14009840500})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:604 +0x78
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func52({0x1091fc4d8, 0x140023dc990}, {0x109228fc0?, 0x14009840500?})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:634 +0x4c
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0x1400067ecc0, {0x108fac6e0?, 0x1?}, 0x14004ac2000)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:179 +0x490
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0x1400b224f78?, {0x1091fc4a0?, 0x10b8abd80?}, 0x140009614a0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x30
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x14000606c60)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x3e8
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x14000606c60, 0x3b9aca00)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x194
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 52519
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1e0
```



### Observed behaviour

panic

### Expected behaviour

no panic

### Steps to reproduce

```bash
replay
```


### Software version

b1cd53abe0fe4403fbbf0b791d4cb315a732cd1f

### Failing test

vega-market-sim amm scenario

### Jenkins run

_No response_

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "0000000000000000000000000000000000000000000000000000000000000000": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "quantum": "1",
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "blockchains.evmBridgeConfigs": "{\"configs\":[{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }]}",
      "blockchains.ethereumRpcAndEvmCompatDataSourcesConfig": "{\"configs\": []}",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "limits.markets.ammPoolEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0003",
      "market.fee.factors.makerFee": "0.00005",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "0000000000000000000000000000000000000000000000000000000000000000",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "0",
      "transfer.fee.maxQuantumAmount": "100",
      "transfer.feeDiscountMinimumTrackedAmount": "0.001",
      "transfer.feeDiscountDecayFraction": "0.5",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1",
      "market.amm.minCommitmentQuantum": "1",
      "market.liquidity.maxAmmCalculationLevels": "100"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: missing min price in levels

goroutine 52521 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x40?, 0x1?, {0x1?, 0x0?, 0x0?})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:196 +0x78
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0x14004a992b0, {0x14009841840, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:262 +0x2c0
go.uber.org/zap.(*Logger).Panic(0x62c45dad955c4ec?, {0x107e3ea1f?, 0x0?}, {0x14009841840, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.25.0/logger.go:258 +0x54
code.vegaprotocol.io/vega/core/matching.(*IndicativePriceAndVolume).getLevelsWithinRange(0x14004ad2120, 0x14006778840, 0x14006778860)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/indicative_price_and_volume.go:318 +0x3a8
code.vegaprotocol.io/vega/core/matching.(*IndicativePriceAndVolume).GetCumulativePriceLevels(0x14004ad2120, 0x104dee014?, 0x140067784e0)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/indicative_price_and_volume.go:346 +0x194
code.vegaprotocol.io/vega/core/matching.(*OrderBook).buildCumulativePriceLevels(0x14002af0e60)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/orderbook.go:536 +0x140
code.vegaprotocol.io/vega/core/matching.(*OrderBook).GetIndicativePrice(0x14002af0e60)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/orderbook.go:448 +0x28
code.vegaprotocol.io/vega/core/matching.(*CachedOrderBook).GetIndicativePrice(0x14002f0eea0)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/matching/cached_orderbook.go:226 +0x64
code.vegaprotocol.io/vega/core/execution/future.(*Market).getAuctionPrice(0x14002af9c00)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/isolated_margin.go:79 +0x94
code.vegaprotocol.io/vega/core/execution/future.(*Market).marginsAuction(0x14002af9c00, {0x1091fc4d8, 0x140023dc990}, 0x14004a99520)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/margin.go:67 +0x168
code.vegaprotocol.io/vega/core/execution/future.(*Market).calcMargins(0x14002af9c00, {0x1091fc4d8, 0x140023dc990}, 0x1400526fb00?, 0x40?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/margin.go:30 +0x58
code.vegaprotocol.io/vega/core/execution/future.(*Market).checkMarginForOrder(0xfffffffffffff243?, {0x1091fc4d8, 0x140023dc990}, 0x1400526fb00?, 0x40?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:3261 +0x28
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitValidatedOrder(0x14002af9c00, {0x1091fc4d8?, 0x140023dc990}, 0x14004a99520)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:2453 +0x468
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitOrder(0x14002af9c00, {0x1091fc4d8, 0x140023dc990}, 0x14004a99520)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:2387 +0x194
code.vegaprotocol.io/vega/core/execution/future.(*Market).SubmitOrderWithIDGeneratorAndOrderID(0x14002af9c00, {0x1091fc4d8?, 0x140023dc990}, 0x40?, {0x1400bee2840, 0x40}, {0x1091cb8c0?, 0x14003ff6090?}, {0x14007341b40, 0x40}, ...)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/future/market.go:2346 +0x50c
code.vegaprotocol.io/vega/core/execution.(*Engine).SubmitOrder(0x14000830c00, {0x1091fc4d8, 0x140023dc990}, 0x1400bca76e0, {0x1400bee2840, 0x40}, {0x1091cb8c0, 0x14003ff6090}, {0x14007341b40, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/engine.go:947 +0x380
code.vegaprotocol.io/vega/core/processor.(*BMIProcessor).ProcessBatch(0x14006761940, {0x1091fc4d8, 0x140023dc990}, 0x14004ac23c0, {0x1400bee2840, 0x40}, {0x14007341980, 0x40}, {0x1092522a8, 0x1400067f740})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/batch_market_instructions_processor.go:237 +0x1040
code.vegaprotocol.io/vega/core/processor.(*App).DeliverBatchMarketInstructions(0x14000a8f500, {0x1091fc4d8, 0x140023dc990}, {0x109228fc0, 0x14009840500}, {0x14007341980, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:1670 +0xfc
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func50({0x1091fc4d8, 0x140023dc990}, {0x109228fc0, 0x14009840500})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:571 +0x64
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).CheckBatchMarketInstructionsW.func51({0x1091fc4d8, 0x140023dc990}, {0x109228fc0, 0x14009840500})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:604 +0x78
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func52({0x1091fc4d8, 0x140023dc990}, {0x109228fc0?, 0x14009840500?})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:634 +0x4c
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0x1400067ecc0, {0x108fac6e0?, 0x1?}, 0x14004ac2000)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:179 +0x490
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0x1400b224f78?, {0x1091fc4a0?, 0x10b8abd80?}, 0x140009614a0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x30
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x14000606c60)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x3e8
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x14000606c60, 0x3b9aca00)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x194
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 52519
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1e0
```

--- a/core/execution/amm/pool.go
+++ b/core/execution/amm/pool.go
@@ -509,7 +509,8 @@ func impliedPosition(sqrtPrice, sqrtHigh num.Decimal, l num.Decimal) *num.Uint {
 // OrderbookShape returns slices of virtual buy and sell orders that the AMM has over a given range
 // and is essentially a view on the AMM's personal order-book.
 func (p *Pool) OrderbookShape(from, to *num.Uint, idgen *idgeneration.IDGenerator) ([]*types.Order, []*types.Order) {
-	buys, sells := []*types.Order{}, []*types.Order{}
+	buys := make([]*types.Order, 0, p.maxCalculationLevels.Uint64())
+	sells := make([]*types.Order, 0, p.maxCalculationLevels.Uint64())
 
 	lower := p.lower.low
 	upper := p.upper.high
@@ -563,15 +564,46 @@ func (p *Pool) OrderbookShape(from, to *num.Uint, idgen *idgeneration.IDGenerato
 
 	var approx bool
 	step := p.oneTick.Clone()
+
 	delta, _ := num.UintZero().Delta(from, to)
 	delta.Div(delta, p.oneTick)
+
+	// we always create accurate orders at the boundary so the approximate region will be two less
+	// because we'll step in one from each end
+	two := num.NewUint(2)
+	if delta.GT(two) {
+		delta.Sub(delta, two)
+	}
 
 	// if there are too many price levels across `from -> to` we have to approximate the orderbook
 	// shape using steps larger than tick size
 	if delta.GT(p.maxCalculationLevels) {
 		step.Div(delta, p.maxCalculationLevels)
+		step.AddSum(num.UintOne()) // if delta / maxcals = 1.9 we're going to want steps of 2
 		step.Mul(step, p.oneTick)
 		approx = true
+
+		// we need to make sure we have an accurate order at the boundaries of the range and we only approximate
+		// steps internally. For example if the AMM is expanded into orders approximately over the interval [100, 200]
+		// in steps of 10, the first order covering the range 100 -> 110 will be priced at ~105. It then looks like
+		// there is no tradable orders at 100. This causes problems in auction uncrossing where we could calculate
+		// an *accurate* crossed region as being [100, 200] but then an *approximate* expansion causes missing volume
+		// at the boundaries.
+		//
+		// We solve this by always creating an accurate order at the boundaries. So for [100, 200] we in-step and
+		// approximate orders over prices [101, 199] then create single orders priced at 100 and 200 exactly.
+
+		bnd := from.Clone()
+		from.Add(from, p.oneTick)
+		to.Sub(to, p.oneTick)
+
+		// create start boundary order
+		o := p.makeBoundaryOrder(bnd, from, fairPrice, idgen)
+		if o.Side == types.SideBuy {
+			buys = append(buys, o)
+		} else {
+			sells = append(sells, o)
+		}
 	}
 
 	ordersFromCurve := func(cu *curve, from, to *num.Uint) {
@@ -648,7 +680,36 @@ func (p *Pool) OrderbookShape(from, to *num.Uint, idgen *idgeneration.IDGenerato
 
 	ordersFromCurve(p.lower, from, to)
 	ordersFromCurve(p.upper, from, to)
+
+	if approx {
+		// create end boundary order
+		bnd := num.UintOne().Add(p.oneTick, to)
+		o := p.makeBoundaryOrder(to, bnd, fairPrice, idgen)
+		if o.Side == types.SideBuy {
+			buys = append(buys, o)
+		} else {
+			sells = append(sells, o)
+		}
+	}
+
 	return buys, sells
+}
+
+func (p *Pool) makeBoundaryOrder(st, nd, fairPrice *num.Uint, idgen *idgeneration.IDGenerator) *types.Order {
+	cu := p.lower
+	if st.GTE(p.lower.high) {
+		cu = p.upper
+	}
+
+	volume := num.DeltaV(
+		cu.positionAtPrice(p.sqrt, st),
+		cu.positionAtPrice(p.sqrt, nd),
+	)
+
+	if nd.GT(fairPrice) {
+		return p.makeOrder(uint64(volume), nd, types.SideSell, idgen)
+	}
+	return p.makeOrder(uint64(volume), st, types.SideBuy, idgen)
 }
 
 // PriceForVolume returns the price the AMM is willing to trade at to match with the given volume of an incoming order.

--- a/core/execution/amm/pool_test.go
+++ b/core/execution/amm/pool_test.go
@@ -47,6 +47,7 @@ func TestOrderbookShape(t *testing.T) {
 	t.Run("test orderbook shape step over fair price", testOrderbookShapeStepOverFairPrice)
 	t.Run("test orderbook shape step fair price at boundary", testOrderbookShapeNoStepOverFairPrice)
 	t.Run("test orderbook shape AMM reduce only", testOrderbookShapeReduceOnly)
+	t.Run("test orderbook shape boundary order when approx", testOrderbookShapeBoundaryOrder)
 }
 
 func testTradeableVolumeInRange(t *testing.T) {
@@ -451,18 +452,18 @@ func testOrderbookShapeLimited(t *testing.T) {
 
 	ensurePosition(t, p.pos, position, num.UintZero())
 	buys, sells := p.pool.OrderbookShape(low, base, nil)
-	assert.Equal(t, 10, len(buys))
+	assert.Equal(t, 11, len(buys))
 	assert.Equal(t, 0, len(sells))
 
 	ensurePosition(t, p.pos, position, num.UintZero())
 	buys, sells = p.pool.OrderbookShape(base, high, nil)
 	assert.Equal(t, 0, len(buys))
-	assert.Equal(t, 10, len(sells))
+	assert.Equal(t, 11, len(sells))
 
 	ensurePosition(t, p.pos, position, num.UintZero())
 	buys, sells = p.pool.OrderbookShape(low, high, nil)
-	assert.Equal(t, 5, len(buys))
-	assert.Equal(t, 5, len(sells))
+	assert.Equal(t, 6, len(buys))
+	assert.Equal(t, 6, len(sells))
 }
 
 func testOrderbookShapeStepOverFairPrice(t *testing.T) {
@@ -477,25 +478,25 @@ func testOrderbookShapeStepOverFairPrice(t *testing.T) {
 	// when we take the step from 24 -> 26 we want to make sure we split that order into two, so we
 	// will actually do maxCalculationLevels + 1 calculations but I think thats fine and keeps the calculations
 	// simple
-	position := int64(7000)
+	position := int64(6000)
 	p.pool.maxCalculationLevels = num.NewUint(10)
 	ensurePosition(t, p.pos, position, num.UintZero())
-	require.Equal(t, "25", p.pool.BestPrice(nil).String())
+	require.Equal(t, "26", p.pool.BestPrice(nil).String())
 
 	ensurePositionN(t, p.pos, position, num.UintZero(), 2)
 	buys, sells := p.pool.OrderbookShape(low, base, nil)
-	assert.Equal(t, 3, len(buys))
+	assert.Equal(t, 4, len(buys))
 	assert.Equal(t, 8, len(sells))
 
 	ensurePosition(t, p.pos, position, num.UintZero())
 	buys, sells = p.pool.OrderbookShape(base, high, nil)
 	assert.Equal(t, 0, len(buys))
-	assert.Equal(t, 11, len(sells))
+	assert.Equal(t, 12, len(sells))
 
 	ensurePositionN(t, p.pos, position, num.UintZero(), 2)
 	buys, sells = p.pool.OrderbookShape(low, high, nil)
-	assert.Equal(t, 2, len(buys))
-	assert.Equal(t, 9, len(sells))
+	assert.Equal(t, 3, len(buys))
+	assert.Equal(t, 10, len(sells))
 }
 
 func testOrderbookShapeNoStepOverFairPrice(t *testing.T) {
@@ -564,6 +565,29 @@ func testOrderbookShapeReduceOnly(t *testing.T) {
 	buys, sells = p.pool.OrderbookShape(base, high, nil)
 	assertOrderPrices(t, buys, types.SideBuy, 10, 11)
 	assert.Equal(t, 0, len(sells))
+}
+
+func testOrderbookShapeBoundaryOrder(t *testing.T) {
+	p := newTestPoolWithRanges(t, num.NewUint(100), num.NewUint(200), num.NewUint(300))
+	defer p.ctrl.Finish()
+
+	midlow := num.NewUint(150)
+	midhigh := num.NewUint(250)
+
+	position := int64(0)
+	ensurePositionN(t, p.pos, position, num.UintZero(), 1)
+
+	// limit the number of orders in the expansion
+	p.pool.maxCalculationLevels = num.NewUint(5)
+
+	buys, sells := p.pool.OrderbookShape(midlow, midhigh, nil)
+	assert.Equal(t, 4, len(buys))
+	assert.Equal(t, 4, len(sells))
+
+	// we're in approximate mode but we still require an exact order at the boundaries of the shape range
+	// check that the price for the first by is midlow, and the last sell is midhigh
+	assert.Equal(t, midlow.String(), buys[0].Price.String())
+	assert.Equal(t, midhigh.String(), sells[(len(sells)-1)].Price.String())
 }
 
 func assertOrderPrices(t *testing.T, orders []*types.Order, side types.Side, st, nd int) {

